miden-tx/src/error.rs

```diff
@@ -166,151 +169,193 @@ impl fmt::Display for AuthenticationError {
+const ERR_FUNGIBLE_ASSET_PROVIDED_ID_IS_INVALID: u32 = 131137;                                      // 131137 0x00020041
+
+// --- NON-FUNGIBLE ASSET ERRORS ------------------------------------------------------------------
+const ERR_NON_FUNGIBLE_ASSET_FORMAT_HIGH_BIT_MUST_BE_ZERO: u32 = 131128;                            // 131128 0x00020038
+const ERR_NON_FUNGIBLE_ASSET_FORMAT_POSITION_ONE_MUST_BE_NON_FUNGIBLE_FAUCET_ID: u32 = 131127;      // 131127 0x00020037
+const ERR_NON_FUNGIBLE_ASSET_ORIGIN_VALIDATION_FAILED: u32 = 131130;                                // 131130 0x0002003A
+const ERR_NON_FUNGIBLE_ASSET_PROVIDED_ID_IS_INVALID: u32 = 131139;                                  // 131139 0x00020043
+
+// --- NOTE ERRORS --------------------------------------------------------------------------------
+const ERR_NOTE_ASSETS_EXCEED_LIMIT_OF_255: u32 = 131114;                                            // 131114 0x0002002A
+const ERR_NOTE_ATTEMPT_TO_ACCESS_NOTE_ASSETS_FROM_INCORRECT_CONTEXT: u32 = 131112;                  // 131112 0x00020028
+const ERR_NOTE_ATTEMPT_TO_ACCESS_NOTE_INPUTS_FROM_INCORRECT_CONTEXT: u32 = 131113;                  // 131113 0x00020029
+const ERR_NOTE_ATTEMPT_TO_ACCESS_NOTE_SENDER_FROM_INCORRECT_CONTEXT: u32 = 131111;                  // 131111 0x00020027
+const ERR_NOTE_DATA_DOES_NOT_MATCH_COMMITMENT: u32 = 131136;                                        // 131136 0x00020040
+const ERR_NOTE_INVALID_TYPE: u32 = 131140;                                                          // 131140 0x00020044
+const ERR_NOTE_TAG_HIGH_BITS_MUST_BE_ZERO: u32 = 131142;                                            // 131142 0x00020046
+const ERR_NOTE_TYPE_TAG_PREFIX_IS_INVALID: u32 = 131141;                                            // 131141 0x00020045
+
+// --- P2ID ERRORS --------------------------------------------------------------------------------
+const ERR_P2ID_EXPECTS_EXACTLY_1_NOTE_INPUT: u32 = 131074;                                          // 131074 0x00020002
+const ERR_P2ID_MISMATCH_OF_TARGET_ACCOUNT_ADDR_AND_TRANSACTION_ADDR: u32 = 131075;                  // 131075 0x00020003
+
+// --- P2IDR ERRORS -------------------------------------------------------------------------------
+const ERR_P2IDR_CAN_BE_RECLAIMED_ONLY_BY_SENDER: u32 = 131077;                                      // 131077 0x00020005
+const ERR_P2IDR_EXPECTS_EXACTLY_2_NOTE_INPUTS: u32 = 131076;                                        // 131076 0x00020004
+const ERR_P2IDR_RECLAIM_BLOCK_HEIGHT_NOT_REACHED: u32 = 131078;                                     // 131078 0x00020006
+
+// --- SWAP ERRORS --------------------------------------------------------------------------------
+const ERR_SWAP_EXPECTS_EXACTLY_9_NOTE_INPUTS: u32 = 131079;                                         // 131079 0x00020007
+const ERR_SWAP_REQUIRES_EXACTLY_1_NOTE_ASSET: u32 = 131080;                                         // 131080 0x00020008
```

Comment:
These are technically not part of the kernel. I think we should break them up into a different category (and maybe use `0x0003` prefix for them).